ExtCore/Collections.HashMap.fs

```diff
@@ -2042,3 +2042,30 @@ module HashMap =
 
         map.MapPartition partitioner
 
+    /// <summary>
+    /// Combines two HashMap's into a single HashMap. Whenever a key exists in both HashMap's, the first HashMaps entry will be added to the result HashMap.
+    /// </summary>
+    /// <param name="map1"></param>
+    /// <param name="map2"></param>
+    /// <returns></returns>
+    [<CompiledName("Union")>]
+    let union (map1 : HashMap<'Key, 'T>) (map2 : HashMap<'Key, 'T>) :  HashMap<'Key, 'T> =
+        checkNonNull "map1" map1
+        checkNonNull "map2" map2
+        
+        match   count map1, count map2 with
```

Comment:
``count`` needs to traverse the map in order to determine the number of key-value pairs in the map. Given you're only concerned with whether the map is empty (or not), please use ``isEmpty`` for that check instead (it's an _O(1)_ operation).